bedevere/util.py

```diff
@@ -1,23 +1,47 @@
 import enum
 import re
 import sys
-from typing import Any, Dict, Union
+from typing import Any, Dict
 
 import gidgethub
 from gidgethub.abc import GitHubAPI
 
-DEFAULT_BODY = ""
-TAG_NAME = f"gh-{{tag_type}}-number"
 NEWS_NEXT_DIR = "Misc/NEWS.d/next/"
-CLOSING_TAG = f"<!-- /{TAG_NAME} -->"
-BODY = f"""\
+PR = 'pr'
+ISSUE = 'issue'
+DEFAULT_BODY = ""
+
+PR_BODY_TAG_NAME = f"gh-{{tag_type}}-number"
+PR_BODY_OPENING_TAG = f"<!-- {PR_BODY_TAG_NAME}: gh-{{pr_or_issue_number}} -->"
+PR_BODY_CLOSING_TAG = f"<!-- /{PR_BODY_TAG_NAME} -->"
+PR_BODY_TEMPLATE = f"""\
 {{body}}
 
-<!-- {TAG_NAME}: gh-{{pr_or_issue_number}} -->
+{PR_BODY_OPENING_TAG}
 * {{key}}: gh-{{pr_or_issue_number}}
-{CLOSING_TAG}
+{PR_BODY_CLOSING_TAG}
+"""
+
+ISSUE_BODY_TAG_NAME = f"gh-linked-{PR}s"
+ISSUE_BODY_OPENING_TAG = f'<!-- {ISSUE_BODY_TAG_NAME} -->'
+ISSUE_BODY_CLOSING_TAG = f'<!-- /{ISSUE_BODY_TAG_NAME} -->'
+ISSUE_BODY_TASK_LIST_TEMPLATE = f"""\n
+{ISSUE_BODY_OPENING_TAG}
+```[tasklist]
+### Linked PRs
+- [ ] gh-{{pr_number}}
+```
+{ISSUE_BODY_CLOSING_TAG}
 """
 
+# Regex pattern to search for tasklist in the issue body
+ISSUE_BODY_TASK_LIST_PATTERN = re.compile(
+    rf"(?P<start>{ISSUE_BODY_OPENING_TAG}\s{{1,2}}```\[tasklist\])"
```

Comment:
`\s{{1,2}}` is used instead of `\n` because when an issue is manually edited GitHub sometimes adds `\r` before `\n` so it becomes `\r\n`

Reply:
```suggestion
    rf"(?P<start>{ISSUE_BODY_OPENING_TAG}\r?\n```\[tasklist\])"
```
Wouldn't this be better/more explicit then?
Alternatively you could also use `[\r\n]+` if you want to allow extra newlines.

Reply:
I feel like this is better, wouldn't want the regex to be too greedy.